### PR TITLE
Restrict subnet auto-discovery to new LB creation on service side

### PR DESCRIPTION
**Pull request by oliviassss**
### Issue

#2056

### Description

Only use subnet auto-discovery when there is no existing Load Balancer on service side.

- When building the scheme, use specified scheme in annotation if any. If there's no such annotation, try to use the scheme in existing Load Balancer. If there's no existing Load Balancer, use "Internal-Facing" as default scheme.
- When building the subnets, resolve subnets by specified subnet name or ID in annotation if any. if there is no such annotation, try to resolve by the subnet name of ID in existing Load Balancer. If there is no existing Load Balancer, then we use auto discovery to resolve subnets.

#### Tests:

- Created service resource, verified the subnet auto-discovery was restricted to the circumstances with no existing load balancers
- Created service resource, verified the build subnet used the subnet Name or ID from annotation directly
- Created service resource, verified the build subnets used the subnet Name or ID from existing Load Balancer when there was no Name or ID in annotation

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/service/model_build_load_balancer.go [hunk 1]
```diff
@@ -217,14 +241,27 @@ func (t *defaultModelBuildTask) getMatchingIPforSubnet(_ context.Context, subnet
...
 		)
 	}
+
+	existingLB, err := t.fetchExistingLoadBalancer(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if existingLB != nil {
+		subnetIDs := make([]string, 0, len(existingLB.LoadBalancer.AvailabilityZones))
+		return t.subnetsResolver.ResolveViaNameOrIDSlice(ctx, subnetIDs,
```
Comment (kishorj) [inline pkg/service/model_build_load_balancer.go:259]: You need to iterate through the subnets in the existingLB.LoadBalancer.AvailabilityZones and push the subnetIDs to the list.

Comment (oliviassss) [reply pkg/service/model_build_load_balancer.go]: Changed, thanks!

**Review by kishorj — APPROVED** [review]

**Pull request merged**